File: production/sheet5/core/debug.py
from pyIsing import IsingModel as IM
import numpy as np
import matplotlib.pyplot as plt
import time
import math

N = 16
eps = 0.01
numLeaps = 20
T = 2
beta = 1/T
print(beta)
ising = IM(N, 100)
number_of_sweeps = 100000


energies = list()
mags = list()
accepted = 0
for i in range(number_of_sweeps):
    print(i)
    if ising.hmc_one_step(beta, numLeaps, eps): # seems to work yay! actually does.
        accepted += 1
    energies.append(ising.E/N/N)
    mags.append(math.sqrt(ising.M_vec_x**2 + ising.M_vec_y**2)/N/N)

print(accepted/number_of_sweeps)

fig , [axE, axM] = plt.subplots(nrows = 2, ncols = 1, sharex = True)
axE.plot(energies)
axM.plot(mags)
plt.show()

File: production/sheet5/core/debug.cpp
#include "core.hpp"
#include <iostream>
#include <vector>
#include <cmath>

int main () {
    unsigned int N = 64;
    IsingModel im(N,0);
    double T = 2.0;
    double delta = 0.1;

    std::vector<double> energy(100000);
    std::vector<double> mag(100000);
    for (int i = 0; i < 100000; i++) {
        im.metropolis_sweep(T, delta);
        energy[i] = im.E;
        mag[i] = std::sqrt(im.M_vec_x*im.M_vec_x + im.M_vec_y*im.M_vec_y);
    }
    

}
